### ISSUE TITLE
[FP] Now and Today values?

### ISSUE THREAD

**Issue by samuelcolburn** [date-fns/date-fns#2531]
I opened a discussion topic similar to this but  that doesn't seem to get much traffic so I'm adding it here as well.

#### Question
how do you use 'now' and 'today' values with the fp functions, such that you always get the current time, especially in a point-free style with other functional libraries like lodash/fp or ramda? 

Since the curried functions expect a Date object, I find myself writing fairly dys-functional (see what I did there) wrappers to stub current date values in order to call them with a current time, like this:
```javascript
import formatRelative from 'date-fns/fp/formatRelative'

export const formatRelativeToNow = (date) => {
  const now = new Date()

  return formatRelative(now)(date)
 }
```

Or writing my own stub methods, and using somewhat awkward compositions like this (using [ramda](https://ramdajs.com/docs/#converge) in this example):
```javascript
const stubNow = () => new Date();

//  quick ramda reference:
//  prop = return object property  
//   converge = ( 'call this with the results of the branching functions', [ array of branching functions'] )
// 
const formatRelativeToNow = compose(
   converge(
    call(formatRelative),
    [stubNow, parseISO]
  ),
  prop("updated_at"), 
);
```
So, has anyone else run into similar issues, and gotten around it in a 'nicer' or 'more functional' way than I'm thinking of? Which leads to the suggestion part.

#### Suggestion
I think it would be nice if: 

- startOfToday was also exported from the `/fp` library just for consistency, possibly aliased as 'stubToday` since it's a common term.
- Add a `stubNow` function
- Add the ability to either pass/recognize these stub functions in the fp functions, such that they always use the current datetime, or add new exports similar to the standard library like `formatRelativeToNow`, `formatDistanceToNow`, etc. 
In other words, I want to be able to write a function like this:
```javascript
const formatDistanceToNow = formatDistance(now)

// Or used in a function composition like
const commentTime = compose(
 formatRelative(now),
 prop('commentDatetime')
)

// Such as in in a React component
const Comment = ({ comment }) => (
 
 <li className="comment">
  <span>{prop('body')}</span>
  <span>{commentTime(comment)}</span>
</li>
)
```

link to [discussion thread](https://github.com/date-fns/date-fns/discussions/2530)
